src/Controller/BannerSelectionController.php

```diff
@@ -67,7 +67,7 @@ private function getCookies( Visitor $visitor, \DateTime $cookieExpirationDate )
 			new Cookie(
 				self::IMPRESSION_COUNT_COOKIE,
 				(string)$visitor->getTotalImpressionCount(),
-				( new \DateTime( 'midnight first day of next month' ) )->modify( '+3 month' )
+				new \DateTime( 'midnight first day of next year' )
```

Comment:
On my machine, PHP interprets that as "current year +1", so it's not January but October (current month). Maybe you should mention January explicitly and also test this cookie date.